### PR TITLE
[3.x] GDNative: Fix `script_data` error when updating placeholder scripts for GDNative libraries

### PR DESCRIPTION
**Pull request by and3rson**
This PR is a backport of https://github.com/godotengine/godot/pull/46514

Tested with latest 3.x branch and several GDNative libraries. Auto-reloading when unfocusing/focusing the editor window still works properly and no error is printed.

Fixes https://github.com/godotengine/godot/issues/42948

### CODE REVIEW

File: modules/gdnative/nativescript/nativescript.cpp [hunk 1]
```diff
@@ -1757,6 +1757,15 @@ void NativeReloadNode::_notification(int p_what) {
 				}
 
 				for (Map<String, Set<NativeScript *>>::Element *U = NSL->library_script_users.front(); U; U = U->next()) {
+					// backported from godotengine/godot#46514
```
Comment (akien-mga) [inline modules/gdnative/nativescript/nativescript.cpp:1760]: That's relevant info for the PR description but not so much for the merged code.

On the other hand, you can give proper credits by adding this to the commit message:
```
Co-authored-by: geekrelief <geekrelief@gmail.com>
```

File: modules/gdnative/nativescript/nativescript.cpp [hunk 2]
```diff
@@ -1757,6 +1757,15 @@ void NativeReloadNode::_notification(int p_what) {
...
+					// multiple gdnative libraries may be reloaded
+					// the gdnative library and script path should match
+					// to prevent failing NSL->library_classes lookup from get_script_desc()
+					// in script->_update_placeholder below
```
Comment (akien-mga) [inline modules/gdnative/nativescript/nativescript.cpp:1764]: Would be good to improve casing and the lines are wrapped a bit early.
```suggestion
					// Multiple GDNative libraries may be reloaded. The library and script
					// path should match to prevent failing `NSL->library_classes` lookup
					// from `get_script_desc()` in `script->_update_placeholder` below.
```
Though comments that refer to code like this inevitably end up outdated when code gets refactored, it's usually better to paraphrase what the code actually does in plain English. But not critical for this to be merged if the fix is accepted.

**Comment by akien-mga** [comment]
Would be nice to amend the commit log to be more explicit about what it does. I tried something for the PR title, feel free to reuse that if that's accurate. The "with comments" part of the PR you're backporting referred to making a new PR with an explanatory comment in the code... it's not the kind of information that should be present in the Git history, it doesn't tell anything about what changed in the engine.

File: modules/gdnative/nativescript/nativescript.cpp (hunk 2 above)
Comment (and3rson) [reply modules/gdnative/nativescript/nativescript.cpp:1764]: Makes sense! I just carbon-copied the original comment for consistency.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by timothyqiu** [comment]
Cherry-picked for 3.5.2
